### PR TITLE
Lookout style analyzer chart

### PR DESCRIPTION
**Pull request by rporres**
Signed-off-by: Rafael Porres Molina <rafa@sourced.tech>

### CODE REVIEW

**Comment by bzz** [comment]
\cc @vmarkovtsev for a quick review

File: lookout-style-analyzer/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - run
+            - lookout.style.format
+            - -s
+            - 0.0.0.0:{{ .Values.app.port }}
```
Comment (bzz) [inline lookout-style-analyzer/templates/deployment.yaml]: This have to be a parameter - it's an address of Lookout server https://github.com/src-d/style-analyzer/blob/master/lookout/core/cmdline.py#L152 and should be configured on deployment to point to `ipv4://lookoutr: 10301` (from https://github.com/src-d/charts/pull/63/files#diff-d50f3a3496ea3f7f7965b6abb4f07ca2R31)

Comment (vmarkovtsev) [reply lookout-style-analyzer/templates/deployment.yaml]: It is also possible to specify a configuration file: https://github.com/src-d/style-analyzer/blob/master/lookout/core/doc/getting_started.md#environment

```
analyzer run lookout.style.format -c config.yml
```

**Comment by vmarkovtsev** [comment]
So there are two addresses: events and data service:

```
--server 0.0.0.0:port1 --request-server 0.0.0.0:port2
```

By default the request server uses the same IP but port 10301.

File: lookout-style-analyzer/templates/deployment.yaml (hunk 1 above)
Comment (rporres) [reply lookout-style-analyzer/templates/deployment.yaml]: what's the difference between `--server` and `--request-server`?

**Comment by rporres** [comment]
i'm kinda lost here now as there are different values in the comments. 

* `--server`
 
The documentation says "Lookout server address, e.g. localhost:1234.", so I understand that we should set it to `lookout:10301`

* `--request-server`

The documentation says "Address of the data retrieval service. \"same\" means --server." 

I have no idea what that means. @bzz comments appear to suggest that we should set it to `lookout:10301` but @vmarkovtsev comment suggest something different

Could you please clarify using as an example a deployment where lookout server is reachable through `lookout:10301`?

**Comment by vmarkovtsev** [comment]
@rporres Alex is correct and I mean the same. This is how everything works:

1. Pull request or Push appears on GitHub
2. Lookout Server learns about it and sends the notification to the Analyzer
3. Analyzer asks the Lookout Server to provide the needed information: UASTs of the changes files, diff, etc.
4. Lookout Server asks Babelfish to parse UASTs and sends the results back to the Analyzer
5. Analyzer does dark magic and sends the review/push comments to the Lookout Server

The address on which the Analyzer listens to notifications from the Lookout Server (step 2) is different from the address the Lookout Server listens to data queries from the Analyzer (step 3).

The step 2's address is `--server`.
The step 3's address is `--request-server`.

**Comment by rporres** [comment]
no, lookout server does not listen in `2000` port. It just listens in 10301.

Port 2000 is the style-analyzer port, so it makes sense that it listens in all addresses, hence the `0.0.0.0`. Taking a look into the container running in staging it confirms that

```
root@lookout-style-analyzer-1947521577-qrnln:/# netstat -na|grep 2000
tcp        0      0 0.0.0.0:2000            0.0.0.0:*               LISTEN
tcp        0      0 10.2.43.202:2000        10.2.61.203:51064       ESTABLISHED
```

From your example, I understand that request-server is `ipv4://lookout:10301`. In our container in staging we see we can reach lookout from style analyzer

```
root@lookout-style-analyzer-1947521577-qrnln:/# nc -vz lookout 10301
lookout.default.svc.cluster.local [10.3.0.52] 10301 (?) open
```

and style analyzer from lookout server

```
root@lookout-3193814884-xbwb3:/# nc -vz lookout-style-analyzer 2000
lookout-style-analyzer.default.svc.cluster.local [10.3.0.179] 2000 (?) open
```

**Pull request merged**